db/sql/a3s/V00__include_pgcrypto_extension.sql

```
@@ -0,0 +1,16 @@
+--
+-- *************************************************
+-- Copyright (c) 2019, Grindrod Bank Limited
+-- License MIT: https://opensource.org/licenses/MIT
+-- **************************************************
+--
+
+DO $$
+BEGIN
+  CREATE EXTENSION IF NOT EXISTS pgcrypto WITH SCHEMA public;
+  COMMENT ON EXTENSION pgcrypto IS 'cryptographic functions';
+EXCEPTION
+  WHEN SQLSTATE '23505' THEN 
+	-- do nothing, the extension is already installed
+END; 
```

I would rethrow the exception for other cases.

---

The exception handler automatically throws any exceptions that are not explicitly dealt with. (What you are requesting happens by default)